library/src/main/java/com/hannesdorfmann/adapterdelegates4/AbsItemAdapterDelegate.java

```diff
@@ -0,0 +1,33 @@
+package com.hannesdorfmann.adapterdelegates4;
+
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+public abstract class AbsItemAdapterDelegate<I extends T, T, VH extends RecyclerView.ViewHolder>
+        extends AdapterDelegate<AdapterItemProvider<T>> {
+
+    @Override
+    protected final boolean isForViewType(@NonNull AdapterItemProvider<T> items, int position) {
+        return isForViewType(items.getAdapterItem(position), items, position);
+    }
+
+    @Override
+    protected final void onBindViewHolder(@NonNull AdapterItemProvider<T> items, int position,
+                                          @NonNull RecyclerView.ViewHolder holder, @NonNull List<Object> payloads) {
+        onBindViewHolder((I) items.getAdapterItem(position), (VH) holder, payloads);
```

Comment:
`items.getAdapterItem(position)` returns nullable value, casting it to non-nullable type `I` can cause a crash.